diff --git a/framework/heightmap.cpp b/framework/heightmap.cpp
index 3a1e9ec..7a3c803 100644
--- a/framework/heightmap.cpp
+++ b/framework/heightmap.cpp
@@ -17,6 +17,7 @@
 
 #include "heightmap.h"
 
+#include <algorithm>
 #include <cstring>
 #include <glm/glm.hpp>
 #include <ktx.h>
diff --git a/framework/scene_graph/scripts/free_camera.h b/framework/scene_graph/scripts/free_camera.h
index 5e9ecf6..7317f39 100644
--- a/framework/scene_graph/scripts/free_camera.h
+++ b/framework/scene_graph/scripts/free_camera.h
@@ -60,7 +60,7 @@ class FreeCamera : public Script
 
 	virtual void resize(uint32_t width, uint32_t height) override;
 
-  private:
+  protected:
 	float speed_multiplier{3.0f};
 
 	glm::vec2 mouse_move_delta{0.0f};
diff --git a/framework/vulkan_sample.cpp b/framework/vulkan_sample.cpp
index bb8b9da..cbf95ac 100644
--- a/framework/vulkan_sample.cpp
+++ b/framework/vulkan_sample.cpp
@@ -121,6 +121,8 @@ VulkanSample::~VulkanSample()
 		device->wait_idle();
 	}
 
+	render_pipeline.reset();
+
 	scene.reset();
 
 	stats.reset();
